crates/miden-tx-batch-prover/src/local_batch_prover.rs

```diff
@@ -8,6 +8,7 @@ use crate::errors::ProvenBatchError;
 
 /// A local prover for transaction batches, proving the transactions in a [`ProposedBatch`] and
 /// returning a [`ProvenBatch`].
+#[derive(Clone, Copy)]
 pub struct LocalBatchProver {
```

Comment:
I would probably just do `Clone` and no `Copy`. This way, in case we make this a heavy struct in the future, it would not be breaking change.